fontforge/fvfonts.c

```diff
@@ -1267,6 +1267,86 @@ return;
 /* *************************** Font Interpolation *************************** */
 /******************************************************************************/
 
+/* In variable fonts, and MM fonts before them, "compatibility" between
+ * contours is needed to acheive a good (or any) result.
+ *
+ * FontForge has no such limitation, and will force contours to be compatible,
+ * then interpolate those. Usually, the result looks horrible.
+ *
+ * So, since the user is probably using this feature to test variable fonts,
+ * ask them if they even want FontForge to force compatibility. */
+bool InterpolationSanity(SplineSet* base, SplineSet *other, int order2, char* prefix) {
+    SplineSet *bss=NULL, *oss=NULL;
+    int bmcc = 0, omcc = 0;
+    bool ret = true;
+
+    if (base == NULL || other == NULL) return false;
+
+    // Count contours...
+    for ( bss = base; bss != NULL; bss=bss->next ) bmcc++;
+    for ( oss = other; oss != NULL; oss=oss->next ) omcc++;
+
+    // An empty glyph is compatible.
+    if (bmcc == 0 && omcc == 0) return true;
+
+    if (prefix == NULL) prefix = "";
+    else prefix = smprintf(_("Glyph %s: "), prefix);
+
+    if (bmcc != omcc) {
+        LogError(_("%sIncompatible contour counts: %d vs %d."), prefix, bmcc, omcc);
+        ret = false;
+    }
+
+    SplinePoint* sp;
+    int contour_count = 1;
+    for ( bss = base, oss = other; bss!=NULL; bss=bss->next, oss=oss->next, contour_count++ ) {
+        if (!ret) break;
+        int bicw = SplinePointListIsClockwise(bss), oicw = SplinePointListIsClockwise(oss);
+
+        if (bicw == -1 || oicw == -1)
+            TRACE(_("%sInterpolationSanity: Assuming self-intersecting contour %d to be sane"), prefix, contour_count);
+        else if (bicw != oicw) {
+            LogError(_("%sIncompatible contour directionality (contour %d)"), prefix, contour_count);
+            ret = false;
+        }
+
+        int bpnum = SplineSetNumberPoints(bss, order2);
```

Comment:
You don't need this anymore, the "after segment" error on (new) line 1326 reports the same info.

You therefore don't need the `order2` parameter (as long as you're not checking quadratic/cubic compatibility at this level).